lib/evmone/eof.cpp

```diff
@@ -98,15 +100,46 @@ std::pair<EOFSectionHeaders, EOFValidationErrror> validate_eof_headers(bytes_vie
     return {section_headers, EOFValidationErrror::success};
 }
 
-std::pair<EOF1Header, EOFValidationErrror> validate_eof1(bytes_view code) noexcept
+EOFValidationErrror validate_instructions(evmc_revision rev, bytes_view code) noexcept
 {
-    const auto [section_headers, error] = validate_eof_headers(code);
-    if (error != EOFValidationErrror::success)
-        return {{}, error};
+    assert(code.size() > 0);  // guaranteed by EOF headers validation
+
+    size_t i = 0;
```

Comment:
Why isn't this an iterator instead?

Reply:
Because with an iterator loop end condition can be only `it != code.end()`, but loop can overrun end in case of truncated immediate.